funnel/models/event.py

```diff
@@ -0,0 +1,148 @@
+
+
+class Participant(BaseMixin, db.Model):
+    """ Model users participating in the proposal space
+        as an attendee, speaker, volunteer, sponsor.
+    """
+    __tablename__ = 'participant'
+
+    fullname = db.Column(db.Unicode(80), nullable=False)
+    #: Unvalidated email address
+    email = db.Column(db.Unicode(80), nullable=False, unique=True)
+    #: Unvalidated phone number
+    phone = db.Column(db.Unicode(80), nullable=True)
+    #: Unvalidated Twitter id
+    twitter = db.Column(db.Unicode(80), nullable=True)
+    #: Job title
+    job_title = db.Column(db.Unicode(80), nullable=True)
+    #: Company
+    company = db.Column(db.Unicode(80), nullable=True)
+    #: Participant's city
+    city = db.Column(db.Unicode(80), nullable=True)
+    # public key
+    puk = db.Column(db.Unicode(44), nullable=False, default=make_public_key, unique=True)
+    #: Access key for connecting to the user record
+    key = db.Column(db.Unicode(44), nullable=False, default=make_private_key, unique=True)
+    badge_printed = db.Column(db.Boolean, default=False, nullable=False)
+    user_id = db.Column(db.Integer, db.ForeignKey('user.id'), nullable=True)
+    user = db.relationship(User, primaryjoin=user_id == User.id,
+        backref=db.backref('participant', cascade="all, delete-orphan"))
+    proposal_space_id = db.Column(db.Integer, db.ForeignKey('proposal_space.id'), nullable=False)
+    proposal_space = db.relationship(ProposalSpace,
+        backref=db.backref('participants', cascade='all, delete-orphan', lazy='dynamic'))
+
+    @classmethod
+    def get_by_event(cls, event, badge_printed=None):
+        participant_attendee_join = db.join(Participant, Attendee, Participant.id == Attendee.participant_id)
+        stmt = db.select([Participant.id, Participant.fullname, Participant.email, Participant.company, Participant.twitter, Participant.puk, Participant.key, Attendee.checked_in]).select_from(participant_attendee_join).where(Attendee.event_id == event.id).order_by(Participant.fullname)
+        if badge_printed:
+            return db.session.execute(stmt.where(Participant.badge_printed == badge_printed)).fetchall()
+        else:
+            return db.session.execute(stmt).fetchall()
+
+
+class Attendee(BaseMixin, db.Model):
+    """ Join model between Participant and Event
+    """
+    __tablename__ = ATTENDEE_TABLE_NAME
+
+    id = db.Column(db.Integer, primary_key=True, unique=True)
+    participant_id = db.Column(None, db.ForeignKey('participant.id'), nullable=False)
+    participant = db.relationship(Participant,
+        backref=db.backref('attendees', cascade='all, delete-orphan', lazy='dynamic'))
+    event_id = db.Column(None, db.ForeignKey('event.id'), nullable=False)
+    event = db.relationship(Event,
+        backref=db.backref('attendees', cascade='all, delete-orphan', lazy='dynamic'))
+    checked_in = db.Column(db.Boolean, default=False, nullable=False)
```

Comment:
Since you have a `checked_in` column, you need the class. This can't be a simple table. `Event.participant` now needs to be an association proxy.